File: 1239-maximum-length-of-a-concatenated-string-with-unique-characters/1239-maximum-length-of-a-concatenated-string-with-unique-characters.py
class Solution:
    def maxLength(self, arr: List[str]) -> int:
        max_len = 0
        def backtrack(index, path):
            nonlocal max_len
            if index >= len(arr):
                if len(set(path)) == len(path):
                    max_len = max(max_len, len(path))
                return
            backtrack(index + 1, path + arr[index])
            backtrack(index + 1, path)
        
        backtrack(0, "")
        return max_len

File: 1239-maximum-length-of-a-concatenated-string-with-unique-characters/1239-maximum-length-of-a-concatenated-string-with-unique-characters.cpp
class Solution {
public:
    
    int max_len = 0;
    
    int maxLength(vector<string>& arr) 
    {
        backtrack("", arr, 0);
        return max_len;
    }
    
    void backtrack(string str, vector <string> &arr, int index)
    {
        if(!isUnique(str)) return;
        if(str.size() > max_len) max_len = str.size();
        for(int i = index; i < arr.size(); i++)
            backtrack(str + arr[i], arr, i+1);
    }
    
    bool isUnique(string str)
    {
        set <char> st;
        for(auto l : str)
        {
            if(st.find(l)!= st.end())
                return false;
            st.insert(l);
        }
        return true;
    }
    
};